modules/shortcodes/instagram.php

```diff
@@ -132,6 +132,79 @@ function jetpack_instagram_embed_reversal( $content ) {
 	return $content;
 }
 
+/**
+ * List of whitelisted and sanitized parameters
+ * that can be used with the Instagram oEmbed endpoint.
+ *
+ * Those parameters can be provided via the Instagram URL, or via shortcode parameters.
+ *
+ * @see https://developers.facebook.com/docs/graph-api/reference/instagram-oembed#parameters
+ *
+ * @since 9.1.0
+ *
+ * @param string $url  URL of the content to be embedded.
+ * @param array  $atts Shortcode attributes.
+ *
+ * @return array $params Array of parameters to be used in Instagram query.
+ */
+function jetpack_instagram_get_whitelisted_parameters( $url, $atts = array() ) {
+	global $content_width;
+
+	// Any URL passed via a shortcode attribute takes precedence.
+	if ( ! empty( $atts['url'] ) ) {
+		$url = $atts['url'];
+		unset( $atts['url'] );
+	}
+
+	/*
+	 * Get URL and parameters from the URL if possible.
+	 *
+	 * We'll also clean any other query params from the URL since Facebook's new API for Instagram
+	 * embeds does not like query parameters. See p7H4VZ-2DU-p2.
+	 */
+	$parsed_url = wp_parse_url( $url );
+	if ( $parsed_url ) {
+		// Bail early if this is not an Instagram URL.
+		if ( ! preg_match( '/instagr(\.am|am\.com)/', $parsed_url['host'] ) ) {
+			return array();
+		}
+
+		$url = 'https://www.instagram.com' . $parsed_url['path'];
+
+		// If we have any parameters as part of the URL, we merge them with our attributes.
+		if ( ! empty( $parsed_url['query'] ) ) {
+			$query_args = array();
+			wp_parse_str( $parsed_url['query'], $query_args );
+
+			$atts = array_merge( $atts, $query_args );
+		}
+	}
+
+	$max_width = 698;
+	$min_width = 320;
+
+	$params = shortcode_atts(
+		array(
+			'url'         => $url,
+			'width'       => isset( $content_width ) ? $content_width : $max_width,
+			'height'      => '',
```

Comment:
Is this list accurate? At https://developers.facebook.com/docs/graph-api/reference/instagram-oembed#parameters, I see no `width` and `height`, only `maxwidth` (and no `maxheight`). It says for `maxwidth`:

> Maximum width of returned media. Must be between 320 and 658. Note that the maxheight parameter is not supported. This is because the embed code is responsive and its height varies depending on its width and length of the caption.

Reply:
Oops, hadn't seen https://github.com/Automattic/jetpack/pull/17700/files#diff-6d4a1998900869c3fd1074d1dc0309783b78409a2e5455930c29c117168261f9R232 :sweat_smile: 

The point about `height` might still be valid tho.

Reply:
`height` is really only used in the AMP view; we don't add it to the Instagram oEmbed query (we only add `url`, `maxwidth`, and `hidecaption`.